core/domain/question_domain.py

```diff
@@ -0,0 +1,145 @@
+        """
+        self.question_id = question_id
+        self.title = title
+        self.question_data = question_data
+        self.data_schema_version = data_schema_version
+        self.collection_id = collection_id
+        self.language_code = language_code
+
+    def to_dict(self):
+        """Returns a dict representing this Question domain object.
+
+        Returns:
+            A dict, mapping all instance of fields of Question instance.
+        """
+        return {
+            'question_id': self.question_id,
+            'title': self.title,
+            'question_data': self.question_data,
+            'data_schema_version': self.data_schema_version,
+            'collection_id': self.collection_id,
+            'language_code': self.language_code
+        }
+
+    def validate(self):
+        """Validates the Question domain object before it is saved."""
+
+        if not isinstance(self.question_id, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.question_id)
+
+        if not isinstance(self.title, basestring):
+            raise utils.ValidationError(
+                'Expected title to be a string, received %s' % self.title)
+
+        if not isinstance(self.question_data, dict):
+            raise utils.ValidationError(
+                'Expected question_data to be a dict, received %s' %
+                self.question_data)
+
+        if not isinstance(self.data_schema_version, int):
+            raise utils.ValidationError(
+                'Expected data_schema_version to be a integer, received %s' %
+                self.data_schema_version)
+
+        if not isinstance(self.collection_id, basestring):
+            raise utils.ValidationError(
+                'Expected collection_id to be a string, received %s' %
+                self.collection_id)
+
+        if not isinstance(self.language_code, basestring):
+            raise utils.ValidationError(
+                'Expected language_code to be a string, received %s' %
+                self.language_code)
+
+        if not any([self.language_code == lc['code']
+                    for lc in constants.ALL_LANGUAGE_CODES]):
+            raise utils.ValidationError(
+                'Invalid language code: %s' % self.language_code)
+
+    @classmethod
+    def from_dict(cls, question_dict):
+        """Returns a Question domain object from dict.
+
+        Returns:
+            Question. The corresponding Question domain object.
+        """
+        question = cls(
+            question_dict['question_id'], question_dict['title'],
+            question_dict['question_data'],
+            question_dict['data_schema_version'],
+            question_dict['collection_id'], question_dict['language_code'])
+
+        return question
+
+    @classmethod
+    def create_default_question(
+            cls, question_id, collection_id,
+            title=feconf.DEFAULT_QUESTION_TITLE,
```

Comment:
Can we actually always pass these in explicitly?

Reply:
We can do it but this is the procedure followed in Collection_domain.py.

Reply:
Let's pass explicitly. Collection_domain was probably just a copy-paste from exploration_domain too.